spatialos-sdk/src/worker/component.rs

```diff
@@ -271,8 +274,20 @@ pub(crate) unsafe fn handle_copy<T>(handle: *mut raw::c_void) -> *mut raw::c_voi
     Arc::into_raw(copy) as *mut _
 }
 
+pub struct WrappedVTable {
+    vtable: Worker_ComponentVtable
+}
+
+impl WrappedVTable {
+    pub fn new(vtable: Worker_ComponentVtable) -> Self {
```

Comment:
Rather than taking the C type as a parameter, I'd suggest having `new` take the component type as a type parameter and then call `create_component_vtable` internally. This simplifies the public interface, and allows us to keep `create_component_vtable` a private implementation detail.

Reply:
Agreed. Could probably just roll `create_component_vtable` directly into `new` as well.

Reply:
That's a good shout